src/state/data-types/actions.js

```diff
@@ -23,7 +39,25 @@ export const fetchDataTypes = (page = { page: 1, pageSize: 10 }, params = '') =>
           dispatch(setDataTypes(json.payload));
           dispatch(toggleLoading('dataTypes'));
           dispatch(setPage(json.metaData));
-        } else if (json && json.errors) {
+        } else {
+          dispatch(addErrors(json.errors.map(err => err.message)));
+          dispatch(toggleLoading('dataTypes'));
+        }
+        resolve();
+      });
+    });
+  })
+);
+
+export const fetchDataType = dataTypeCode => dispatch => (
+  new Promise((resolve) => {
+    dispatch(toggleLoading('dataTypes'));
+    getDataType(dataTypeCode).then((response) => {
+      response.json().then((json) => {
+        if (response.ok) {
+          dispatch(setDataTypes([json.payload]));
+          dispatch(initialize('dataType', json.payload));
```

Comment:
Please call  `dispatch(toggleLoading('dataTypes'));` outside the if/else block

Reply:
done